### DISCUSSION TITLE
🚀 Feature Request: Query Versions by tag

### DISCUSSION THREAD

**Discussion by sayantank** [cloudflare/workers-sdk#13500] [Feature Requests]
### Describe the solution

Currently, there is only way to query a version, which is by its `version_id`. It would be pretty nice if I could query a version by its `tag`.

The use-case I would like this is for, I could tag every deployment with the git tag for my workers, and then have my CI pipeline easily query them to have a configurable way to run deployments.

**Comment by Mrinank-Bhowmick** [comment]
Hey @sayantank ,
I would like to work on this issue. I read the api docs and found out that we can get version details using {version_id} only. [docs](https://developers.cloudflare.com/api/operations/worker-versions-get-version-detail)
Would it be ok if we fetch results by version_id and use `.find(...)` in js and iterates through the response array and return
